Tiltfile

```diff
@@ -0,0 +1,190 @@
+# -*- mode: Python -*-
+
+k8s_yaml([
+'config/crd/bases/velero.io_backups.yaml',
+'config/crd/bases/velero.io_backupstoragelocations.yaml', 
+'config/crd/bases/velero.io_deletebackuprequests.yaml',   
+'config/crd/bases/velero.io_downloadrequests.yaml',   
+'config/crd/bases/velero.io_podvolumebackups.yaml',  
+'config/crd/bases/velero.io_podvolumerestores.yaml',  
+'config/crd/bases/velero.io_resticrepositories.yaml',
+'config/crd/bases/velero.io_restores.yaml',    
+'config/crd/bases/velero.io_schedules.yaml', 
+'config/crd/bases/velero.io_serverstatusrequests.yaml',  
+'config/crd/bases/velero.io_volumesnapshotlocations.yaml', 
+])
+
+# default values
+settings = {
+   "default_registry": "",
+   "enable_restic": False,
+   "create_backup_locations": False, 
+   "setup-minio": False, 
+}
+
+# global settings
+settings.update(read_json(
+    "tilt-resources/tilt-settings.json",
+    default = {},
+))
+
+k8s_yaml(kustomize('tilt-resources'))
+k8s_yaml('tilt-resources/deployment.yaml')
+if settings.get("enable_restic"):
+    k8s_yaml('tilt-resources/restic.yaml')
+if settings.get("create_backup_locations"):
+    k8s_yaml('tilt-resources/velero_v1_backupstoragelocation.yaml')
+if settings.get("setup-minio"):
+    k8s_yaml('examples/minio/00-minio-deployment.yaml', allow_duplicates=True)
+
+# By default, Tilt automatically allows Minikube, Docker for Desktop, Microk8s, Red Hat CodeReady Containers, Kind, K3D, and Krucible.
+allow_k8s_contexts(settings.get("allowed_contexts"))
+default_registry(settings.get("default_registry"))
+
+tilt_helper_dockerfile_header = """
+# Tilt image
+FROM golang:1.15.3 as tilt-helper
+# Support live reloading with Tilt
+RUN wget --output-document /restart.sh --quiet https://raw.githubusercontent.com/windmilleng/rerun-process-wrapper/master/restart.sh  && \
+    wget --output-document /start.sh --quiet https://raw.githubusercontent.com/windmilleng/rerun-process-wrapper/master/start.sh && \
+    chmod +x /start.sh && chmod +x /restart.sh
+"""
+
+additional_docker_helper_commands = """
+RUN wget -qO- https://dl.k8s.io/v1.19.2/kubernetes-client-linux-amd64.tar.gz | tar xvz
+RUN wget -qO- https://get.docker.com | sh
+"""
+
+additional_docker_build_commands = """
+COPY --from=tilt-helper /usr/bin/docker /usr/bin/docker
+COPY --from=tilt-helper /go/kubernetes/client/bin/kubectl /usr/bin/kubectl
+"""
+
+##############################
+# Setup Velero
+##############################
+
+# Set up a local_resource build of the Velero binary. The binary is written to .tiltbuild/velero.
+local_resource(
+    "velero_manager",
+    cmd = "cd " + "." + ';mkdir -p .tiltbuild;PKG=github.com/vmware-tanzu/velero BIN=velero GOOS=linux GOARCH=amd64 VERSION=main GIT_SHA=aa4730970037ce29 GIT_TREE_STATE=dirty OUTPUT_DIR=.tiltbuild ./hack/build.sh',
```

Comment:
If this works, I think we could update the `build.sh` script to make the `GIT_SHA` optional, _or_ look up the value in the Tiltfile once at the start.

That way, the value's not hardcoded to something that may confuse a dev looking at their binary someday and wondering why the built SHA doesn't match what they're actually building.

Reply:
This worked. I have to do the same with the plugin bit.

Ok, I'll changed the build.sh. 

Should `PKG` be set to `github.com/vmware-tanzu/velero` or to where the local src code is?

Reply:
I'm pointing the PKG variable to local code. Let me know if it should be different.